src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -646,16 +646,44 @@ bool VU_TASKS_QUEUE_FLAGS[VU_TASK_TYPES_SIZE] = {0};
 
 void vu_tasks_queue_push_callback (void* data) {
     if (data) {
-        VU_TASKS_QUEUE_FLAGS[((vu_task_context_t*)data)->task_type] = true;
+        VU_TASKS_QUEUE_FLAGS[((vu_task_ctx_t*)data)->task_type] = true;
     }
 }
 
 void vu_tasks_queue_pop_callback (void* data) {
     if (data) {
-        VU_TASKS_QUEUE_FLAGS[((vu_task_context_t*)data)->task_type] = false;
+        VU_TASKS_QUEUE_FLAGS[((vu_task_ctx_t*)data)->task_type] = false;
     }
 }
 
+vu_task_ctx_t *create_task(const char *task_parameters,...) {
```

Comment:
I propose that all the fields in `vu_task_ctx_t` have a value even when they aren't used. 

Maybe we could avoid the creation of the local variables and store the result of `va_arg()` in the structure.

And if the parameters **id**, **type** and **status** are mandatory maybe they could be out of the variable argument list.

Reply:
Changed!